doc/hardware.rst

```diff
@@ -0,0 +1,426 @@
+4          SDA    Data         P01-3    GPIO 2 (SDA)
+========== ====== ============ ======== ==============
+
+You can also solder the wires directly to the underside of the RPi GPIO pins.
+
+.. seealso::
+
+  Alternatively, on rev.2 RPi's, right next to the male pins of the P1 header,
+  there is a bare P5 header which features I2C channel 0, although this doesn't
+  appear to be initially enabled and may be configured for use with the Camera
+  module.
+
+  +------------+------+-------------+---------+---------------+-------------------------------------+
+  | Device Pin | Name | Remarks     | RPi Pin | RPi Function  | Location                            |
+  +============+======+=============+=========+===============+=====================================+
+  | 1          | GND  | Ground      | P5-07   | GND           | .. image:: images/RPi_P5_header.png |
+  +------------+------+-------------+---------+---------------+                                     |
+  | 2          | VCC  | +3.3V Power | P5-02   | 3V3           |                                     |
+  +------------+------+-------------+---------+---------------+                                     |
+  | 3          | SCL  | Clock       | P5-04   | GPIO 29 (SCL) |                                     |
+  +------------+------+-------------+---------+---------------+                                     |
+  | 4          | SDA  | Data        | P5-03   | GPIO 28 (SDA) |                                     |
+  +------------+------+-------------+---------+---------------+-------------------------------------+
+
+Ensure that the I2C kernel driver is enabled::
+
+  $ dmesg | grep i2c
+  [    4.925554] bcm2708_i2c 20804000.i2c: BSC1 Controller at 0x20804000 (irq 79) (baudrate 100000)
+  [    4.929325] i2c /dev entries driver
+
+or::
+
+  $ lsmod | grep i2c
+  i2c_dev                 5769  0
+  i2c_bcm2708             4943  0
+  regmap_i2c              1661  3 snd_soc_pcm512x,snd_soc_wm8804,snd_soc_core
+
+If you have no kernel modules listed and nothing is showing using ``dmesg``
+then this implies the kernel I2C driver is not loaded. Enable the I2C as
+follows:
+
+#. Run ``sudo raspi-config``
+#. Use the down arrow to select ``5 Interfacing Options``
+#. Arrow down to ``P5 I2C``
+#. Select **yes** when it asks you to enable I2C
+#. Also select **yes** when it asks about automatically loading the kernel module
+#. Use the right arrow to select the **<Finish>** button
+
+After rebooting re-check that the ``dmesg | grep i2c`` command shows whether
+I2C driver is loaded before proceeding. You can also
+`enable I2C manually <http://elinux.org/RPiconfig#Device_Tree>`_ if the
+``raspi-config`` utility is not available.
+
+Optionally, to improve performance, increase the I2C baudrate from the default
+of 100KHz to 400KHz by altering ``/boot/config.txt`` to include::
+
+  dtparam=i2c_arm=on,i2c_baudrate=400000
+
+Then reboot.
+
+Next, add your user to the *i2c* group and install ``i2c-tools``::
```

Comment:
i would use double backticks for i2c

Reply:
This was copied from the luma.oled content.  If we change it here, probably should change it in both places.

Reply:
make sense (to start here at least)